ArduCopter/control_guided.cpp

```diff
@@ -686,6 +694,8 @@ void Copter::guided_set_desired_velocity_with_accel_and_fence_limits(const Vecto
 #if AC_AVOID_ENABLED
     // limit the velocity to prevent fence violations
     avoid.adjust_velocity(pos_control->get_pos_xy_kP(), pos_control->get_accel_xy(), curr_vel_des);
+    // get avoidance adjusted climb rate
+    curr_vel_des.z = get_avoidance_adjusted_climbrate(curr_vel_des.z);    
```

Comment:
It's up to @rmackay9, but I would directly do ` avoid.adjust_velocity_z(pos_control->get_pos_z_kP(), pos_control->get_accel_z(), curr_vel_des.z);`.

The `get_avoidance_adjusted_climbrate` is useful for other flight modes where we don't need to put macro checks, here we already have the macro check.

Reply:
Yes @OXINARF , you are absolutely right about this. My bad, I should have seen this.